File: Data Structures & Algorithms/is-palindrome/submission-0.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();

        int start = 0;
        int end = s.length() - 1;
        
        while(start <= end){
            while(start < s.length() && !((s.charAt(start) >= 97 && s.charAt(start) <= 122) || (s.charAt(start) >= 48 && s.charAt(start) <= 57))){
                start++;
            }
            while(end >= 0 && !((s.charAt(end) >= 97 && s.charAt(end) <= 122) || (s.charAt(end) >= 48 && s.charAt(end) <= 57))){
                end--;
            }

            if(start > end){
                break;
            }

            if(s.charAt(start) != s.charAt(end)){
                return false;
            }
            start++;
            end--;
        }

        return true;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        n = len(nums)

        def backtrack(start, curr_list):
            res.append(curr_list.copy())

            for i in range(start, n):
                curr_list.append(nums[i])
                backtrack(i + 1, curr_list)
                curr_list.pop()

        backtrack(0, [])
        return res